src/liballoc/collections/binary_heap.rs

```diff
@@ -568,7 +568,7 @@ impl<T: Ord> BinaryHeap<T> {
             while child < end {
                 let right = child + 1;
                 // compare with the greater of the two children
-                if right < end && !(hole.get(child) > hole.get(right)) {
+                if right < end && (hole.get(child) <= hole.get(right)) {
```

Comment:
Same here, unnecessary parentheses.

Reply:
I guess the `unused-parens` lint only warns on outermost parenthesis?